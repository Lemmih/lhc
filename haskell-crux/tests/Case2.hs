module Case2 where

x = case x of
      x -> x

y :: ()
y = case y of
      y -> y

z :: ()
z = case z of
      () -> ()
