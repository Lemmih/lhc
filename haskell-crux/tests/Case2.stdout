Case2.x : ∀ v0. v0
Case2.x =
  λ @.v0^1:* → 
  Case2.x @.v0^1
Case2.y : ()
Case2.y =
  Case2.y
Case2.z : ()
Case2.z =
  scrut^2 ← Case2.z
  LHC.Prim.Unit  ← scrut^2:()
  LHC.Prim.Unit
